deepchecks/checks/performance/model_error_analysis.py

```diff
@@ -0,0 +1,317 @@
+                                                                              data[feature].isin(weak_categories))
+
+                    color_map[weak_name_feature] = weak_color
+                else:
+                    weak_name_feature = None
+
+                replace_dict = {x: weak_name_feature if x in weak_categories else ok_name_feature for x in
+                                error_per_segment_ser.index}
+                color_col = data[feature].replace(replace_dict)
+
+                # Display
+                display.append(px.violin(
+                    data, y=error_col_name, x=feature, title=f'Segmentation of error by {feature}', box=False,
+                    labels={error_col_name: 'model error'}, color=color_col,
+                    color_discrete_map=color_map
+                ))
+            else:
+                data = data.sample(n_samples_display, random_state=self.random_state)
+
+                # Train tree to partition segments according to the model error
+                tree_partitioner = DecisionTreeRegressor(max_depth=1,
+                                                         min_samples_leaf=self.min_segment_size + np.finfo(float).eps,
+                                                         random_state=self.random_state).fit(
+                    data[[feature]], data[error_col_name])
+                if len(tree_partitioner.tree_.threshold) > 1:
+                    threshold = tree_partitioner.tree_.threshold[0]
+                    color_col = data[feature].ge(threshold)
+
+                    segment1_text, segment1_details = get_segment_details(model, scorer, test_dataset, data,
+                                                                          color_col)
+                    segment2_text, segment2_details = get_segment_details(model, scorer, test_dataset, data,
+                                                                          ~color_col)
+                    color_col = color_col.replace([True, False], [segment1_text, segment2_text])
+                    if segment1_details['score'] >= segment2_details['score']:
+                        color_map = {segment1_text: ok_color, segment2_text: weak_color}
+                    else:
+                        color_map = {segment1_text: weak_color, segment2_text: ok_color}
+                else:
+                    color_col = data[error_col_name]
+                    color_map = None
+                display.append(px.scatter(data, x=feature, y=error_col_name, color=color_col,
+                                          title=f'Segmentation of error by {feature}',
+                                          labels={error_col_name: 'model error'},
+                                          color_discrete_map=color_map))
+
+            value[feature] = {'segment1': segment1_details or None, 'segment2': segment2_details or None}
```

Comment:
Why not name segments as "high error" and "low error" segments? Will also help with max() in the condition

Reply:
At this stage, I don't know yet if it's high or low. Need to check in that if.

Reply:
Yes, but why not insert it into "low" and "high" to begin with when you're doing the tree_partitioner thing? why keep it as segment1 and 2? for what benefit?

Reply:
oh now I get it, yeah could have changed it here...